Disable autosave when uploading

will-byrne opened the pull request:
#### Background

Disable autosave while files are uploading

- [x] Auto save and save on next disabled on upload
- [x] Trigger save if form changes made during upload then upload completes
- [x] `fileStatus` validation working on going back then next while upload is in progress
- [x] Auto save disabled when submitting
- [ ] Ensure `lastStepVisited` is set to the author step when first creating the manuscript
#### Any relevant tickets

Closes #2051

erezmus committed:
Disable autosave when uploading

will-byrne committed:
remove console logging

petereast reviewed (approved):
Sure

will-byrne committed:
add extra logging for race condition check

erezmus committed:
- Enqueue requests for updateManuscript
- Removed savePage mutation
- Removed use of SavePageStatus

erezmus committed:
omit lastStepVisited from manuscript input for final submission

erezmus committed:
generalised the queueing of requests and made it return promises so that its thenable and fixes going to thank you page

erezmus committed:
Merge branch 'develop' into 2051_disable_autosave_on_upload

Commit:
Add suspendSave state to SubmissionWizard.

Commit:
Typo fix for disableSave

Cooryd commented:
We've got autosave and save on next disabled during uplaod now. This has introduced a bug with validation of `fileStatus` when going back then next when something is uploading that needs addressing

Cooryd commented:
If this issue causes too many issues, we may need to disable 'back' as well as 'next' navigation while files upload.

Cooryd commented on packages/component-submission/client/components/WizardStep.js:
```diff
@@ -48,7 +47,8 @@ class WizardStep extends React.Component {
         key={title}
         onSubmit={values => {
-          handleButtonClick(values).then(() => history.push(nextUrl))
+          handleButtonClick({ ...values, lastStepVisited: nextUrl })
```
This was my own change but will moving the `history.push` out of the resolved promise cause any issues when the form does need to save when navigating? will we lose the last 5 seconds of data?

erezmus committed:
moved file uploading state into SubmissionWizard, disable back button during upload operation

petereast committed:
Merge save and disable upload & add tests

petereast committed:
\'then\' not \'finally\'

petereast commented:
Okay, so now there's autosave weirdness - I had problems with changes I made being reset on the <kbd>Author</kbd> and <kbd>Details</kbd> pages.

Thank you, _browser tests_ - that could've been nasty :1st_place_medal:

will-byrne committed:
prevent autosave from triggering while submiting

- prevent autosave from triggering during submission
- promisified onNextClick to allow the save to return before loading the next page

Cooryd commented on packages/component-submission/client/pages/FilesStepPage.js:
```diff
@@ -76,8 +74,7 @@ export class FilesStepPageComponent extends React.Component {
       })
-      .finally(() => {
-        disableSave(false)
+      .then(() => {
```
We'll be stuck in a state of `isUploading` if this throws an error. This will then cause the client to be stuck in the converting stater so the error will be hidden

will-byrne committed:
Remove old page status component and fix review issue
